packages/babel-plugin-proposal-pattern-matching/src/index.js

```diff
@@ -0,0 +1,312 @@
+import syntaxPatternMatching from "@babel/plugin-syntax-pattern-matching";
+
+export default function({ types: t }) {
+  function makeIsArrayTest(body) {
+    return t.callExpression(
+      t.memberExpression(t.identifier("Array"), t.identifier("isArray"), false),
+      [body],
+    );
+  }
+
+  function makeArrayLengthTest(id, length) {
+    return t.binaryExpression(
+      "===",
+      t.memberExpression(id, t.identifier("length")),
+      t.numericLiteral(length),
+    );
+  }
+
+  function makeIsObjectTest(body) {
+    return t.binaryExpression(
+      "===",
+      t.unaryExpression("typeof", body, true),
+      t.stringLiteral("object"),
+    );
+  }
```

Comment:
You don't have to do this but you should be able to use babel-template instead of creating the node manually for a lot of these https://github.com/babel/babel/blob/master/packages/babel-template/README.md

Reply:
@hzoo Thanks for suggessions. Updated in the latest commit.